src/vaadin-combo-box-mixin.html

```diff
@@ -580,16 +580,35 @@
      */
     _onEscape(e) {
       if (this.autoOpenDisabled) {
-        this._focusedIndex = -1;
-        this.cancel();
-      } else if (this.opened) {
-        this._stopPropagation(e);
-
-        if (this._focusedIndex > -1) {
+        // Auto-open is disabled
+        if (this.value !== this._inputElementValue && this._inputElementValue.length > 0) {
+          // The input value has changed but the change hasn't been committed, so cancel it.
+          this._stopPropagation(e);
           this._focusedIndex = -1;
-          this._revertInputValue();
-        } else {
           this.cancel();
+        } else if (this.clearButtonVisible && !this.opened && !!this.value) {
+          this._stopPropagation(e);
+          // The clear button is visible and the overlay is closed, so clear the value.
+          this._clear();
```

Comment:
The PR has a regression which should be covered in the `_onEscape` logic: if you open the dropdown of an `autoOpenDisabled` combobox, pressing esc now also clears the value

https://user-images.githubusercontent.com/1222264/141256666-b0ba2c40-c364-44dc-ba5d-b7a49e88937e.mp4

Reply:
Curious. I added a test for that specific scenario and it's passing. I must check it.

Reply:
Ah, the value of the combo-box doesn't change but the value of the internal text-field does.

Reply:
Added an extra condition to the first check on autoOpenDisabled instances.